src/silo/storage/column/insertion_index.cpp

```diff
@@ -0,0 +1,235 @@
+#include "silo/storage/column/insertion_index.h"
+
+#include <algorithm>
+#include <cassert>
+#include <iterator>
+#include <string_view>
+#include <unordered_set>
+#include <utility>
+
+#include <boost/functional/hash.hpp>
+
+#include "silo/common/numeric_conversion.h"
+#include "silo/common/string_utils.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+
+namespace silo::storage::column::insertion {
+
+namespace {
+
+constexpr std::string_view DELIMITER_INSERTIONS = ",";
+constexpr std::string_view DELIMITER_INSERTION = ":";
+constexpr std::string_view REGEX_ANY = ".*";
+
+std::pair<uint32_t, std::string> parseInsertion(const std::string& value) {
+   const auto position_and_insertion = splitBy(value, DELIMITER_INSERTION);
+   if (position_and_insertion.size() != 2) {
+      const std::string message = "Failed to parse insertion due to invalid format: " + value;
+      throw PreprocessingException(message);
+   }
+   const auto position = tryConvertStringToU32(position_and_insertion[0]);
+   const auto& insertion = position_and_insertion[1];
+   return std::make_pair(position, insertion);
+}
+
+struct ThreeMerHash {
+   size_t operator()(const InsertionIndex::three_mer_t& three_mer) const {
+      size_t seed = 0;
+      for (const auto one_mer : three_mer) {
+         boost::hash_combine(seed, std::hash<NUCLEOTIDE_SYMBOL>{}(one_mer));
+      }
+      return seed;
+   }
+};
+
+std::vector<InsertionIndex::three_mer_t> extractThreeMers(const std::string& search_pattern) {
+   std::unordered_set<InsertionIndex::three_mer_t, ThreeMerHash> result;
+   for (const auto& continuous_string : splitBy(search_pattern, REGEX_ANY)) {
+      auto continuous_symbols = stringToNucleotideSymbolVector(continuous_string);
+      if (continuous_symbols == std::nullopt) {
+         throw std::runtime_error("Wrong symbol in pattern: " + continuous_string);
```

Comment:
It would actually be nice to know which symbol was wrong.

Reply:
Adapted it